packages/babel-core/src/transformation/util/missing-plugin-helper.js

```diff
@@ -0,0 +1,129 @@
+    syntax: "@babel/plugin-syntax-do-expressions",
+    transform: "@babel/plugin-proposal-do-expressions",
+  },
+  dynamicImport: {
+    syntax: "@babel/plugin-syntax-dynamic-import",
+  },
+  // Not required to be handled.
+  // (See https://github.com/babel/babel/issues/6205#issuecomment-346997662).
+  // Here just to complete the plugin map
+  estree: {},
+  exportDefaultFrom: {
+    syntax: "@babel/plugin-syntax-export-default-from",
+    transform: "@babel/plugin-proposal-export-default-from",
+  },
+  exportNamespaceFrom: {
+    syntax: "@babel/plugin-syntax-export-namespace-from",
+    transform: "@babel/plugin-proposal-export-namespace-from",
+  },
+  flow: {
+    syntax: "@babel/plugin-syntax-flow",
+    transform: "@babel/plugin-transform-flow-strip-type",
+  },
+  functionBind: {
+    syntax: "@babel/plugin-syntax-function-bind",
+    transform: "@babel/plugin-proposal-function-bind",
+  },
+  functionSent: {
+    syntax: "@babel/plugin-syntax-function-sent",
+    transform: "@babel/plugin-proposal-function-sent",
+  },
+  importMeta: {
+    issueLink: "https://github.com/babel/proposals/issues/10",
+  },
+  jsx: {
+    syntax: "@babel/plugin-syntax-jsx",
+    transform: "@babel/plugin-transform-react-jsx",
+  },
+  nullishCoalescingOperator: {
+    syntax: "@babel/plugin-syntax-nullish-coalescing-operator",
+    transform: "@babel/plugin-proposal-nullish-coalescing-operator",
+  },
+  numericSeparator: {
+    syntax: "@babel/plugin-syntax-numeric-separator",
+    transform: "@babel/plugin-proposal-numeric-separator",
+  },
+  objectRestSpread: {
+    syntax: "@babel/plugin-syntax-object-rest-spread",
+    transform: "@babel/plugin-proposal-object-rest-spread",
+  },
+  optionalCatchBinding: {
+    syntax: "@babel/plugin-syntax-optional-catch-binding",
+    transform: "@babel/plugin-proposal-optional-catch-binding",
+  },
+  optionalChaining: {
+    syntax: "@babel/plugin-syntax-optional-chaining",
+    transform: "@babel/plugin-proposal-optional-chaining",
+  },
+  pipelineOperator: {
+    syntax: "@babel/plugin-syntax-pipeline-operator",
+    transform: "@babel/plugin-proposal-pipeline-operator",
+  },
+  throwExpressions: {
+    syntax: "@babel/plugin-syntax-throw-expressions",
+    transform: "@babel/plugin-proposal-throw-expressions",
+  },
+  typescript: {
+    syntax: "@babel/plugin-syntax-typescript",
+    transform: "@babel/plugin-transform-typescript",
+  },
+};
+
+export default function generateHelpMessage(
+  missingPlugins: string[] = [],
+): string {
+  let helpMessage = "";
+  const pluginName = missingPlugins.length && missingPlugins[0];
+  if (pluginName) {
+    const pluginInfo = pluginNameMap[pluginName];
+    if (pluginInfo) {
+      const {
+        syntax: syntaxPlugin,
+        transform: transformPlugin,
+        issueLink,
+      } = pluginInfo;
+      if (syntaxPlugin) {
+        if (transformPlugin) {
+          helpMessage =
+            `Use the ${syntaxPlugin} plugin to enable parsing. ` +
```

Comment:
What about adding `This experimental syntax isn't enabled by default` as an "inroduction" to the error message?

Reply:
I'm wondering if we even need a different message in the case where there's only a syntax plugin and not a transform (just dynamic-import)? I feel like we can just simplify this down to something like:

> Support for this experimental syntax isn't currently enabled. Add `<insert transform/syntax plugin here>` to the `plugins` section of your Babel config to enable it.

(Also, do we need the line re: syntax plugin is enabled by the transform? I feel like most people don't even know the difference?)

Reply:
I would also add a link to the website because we can explain the user how to install it, and if a transformation is needed. What do you think? That also means we should have doc about all the proposals (or issues if not official)

Reply:
I'll use @existentialism 's messages. I think they take care of @nicolo-ribaudo 's suggestion as well. 
@xtuc like you mentioned, not all the plugins have docs on the website. Would it be better to point to the particular package's README instead?

Reply:
@maaz93 Yes it is. Since it's a link it's generic enough, we can change it later if needed.

Reply:
@xtuc Let me clear my understanding. So, if I was suggesting a message for missing `asyncGenerators` plugin, I would have to add `https://github.com/babel/babel/tree/master/packages/babel-plugin-syntax-async-generators` and `https://github.com/babel/babel/tree/master/packages/babel-plugin-proposal-async-generator-functions` in the message? Or I need to suggest `https://babeljs.io/docs/plugins/` for more info?

Reply:
I can use the READMEs for now, I'm not sure that we have the complete doc on the website tho.

Reply:
While trying to add this, I realised that the message will get very long with links in it. I thought I'll just suggest

> You can find more details about every plugin at https://github.com/babel/babel/tree/master/packages

Will this be enough?

Reply:
We could even use git.io and make it shorter

Reply:
@nicolo-ribaudo That was a great suggestion! I wrote a quick dirty script to use a [gitio](https://www.npmjs.com/package/gitio) in a loop and generate short URLs. The map looks something like this now

```javascript
asyncGenerators: {
  syntax: {
    name: "@babel/plugin-syntax-async-generators",
    url: "https://git.io/vb4SY",
  },
  transform: {
    name: "@babel/plugin-proposal-async-generator-functions",
    url: "https://git.io/vb4yp",
  },
}
```
With this, the message would look like

> Support for this experimental syntax isn't currently enabled. Add either `<insert transform plugin here>`(`insert URL`) or `<insert syntax plugin>`(`insert URL`) (if you only need parsing support) to the plugins section of your Babel config.

Is this more clear?